client/core/core.go

```diff
@@ -3067,9 +3106,11 @@ func (c *Core) resumeTrades(dc *dexConnection, trackers []*trackedTrade) assetMa
 			notifyErr("Wallet missing", "Wallet retrieval error for active order %s: %v", tracker.token(), err)
 			continue
 		}
+
 		tracker.mtx.Lock()
 		tracker.wallets = wallets
 		tracker.mtx.Unlock()
```

Comment:
There are a couple other sites where similar lock appears necessary - e.g. when setting `match.failErr` and `tracker.coins`; and when reading some tracker fields.

Since this tracker isn't yet added to the dc.trades map at this point, the mutex locks may not be necessary, but if that's the thinking, let's remove this lock as well to be consistent.

`tracker.unspentContractAmounts()` now requires mutex RLock; and I made some changes below to satisfy this requirement but I'm recommending performing required mutex locks in this method everywhere necessary or nowhere at all.

I previously tried to hold this mutex lock for longer in c8079a3cd66ec8bb938a2a2fbee87d6fe65c2ec1, but that led to a deadlock, fixing which, would require some refactoring.

Reply:
> Since this tracker isn't yet added to the dc.trades map at this point

That does seem to be the case.  No need for locking if it cannot be accessed concurrently.